### DISCUSSION TITLE
Local RAG - which embedding to use?

### DISCUSSION THREAD

**Discussion by ianscrivener** [ggml-org/llama.cpp#3518] [Q&A] [answered]
I'm coding a RAG demo with llama.cpp, [Weaviate](https://weaviate.io/) vector database and [LlamaIndex](https://www.llamaindex.ai/).. (which works closely with langchain).

Should I use llama.cpp embeddings, or a leading embedding model like [BAAI/bge-small-en](https://huggingface.co/BAAI/bge-small-en)?

**Comment by Alumniminium** [comment, marked as answer]
i use bge-large-en-1.5 and it works very well.

**Reply by ianscrivener** [reply]
thanks @Alumniminium

**Reply by moatftw** [reply]
May I ask how do you use bge-large-en-1.5 in llama.cpp? Since I can't find it in the GGUF format that can be loaded in llama.cpp's embedding example.

**Reply by ianscrivener** [reply]
@Alumniminium ?

**Reply by Alumniminium** [reply]
Im not using it in llama.cpp im using it together with llama.cpp.
